### PR TITLE
Datatypes for electric vehicles

### LINKED ISSUE #4: Create data structures
- [x] charging points (supply)
- [ ] electric vehicles (demand)

### PR DESCRIPTION
**Pull request by mdavezac**
Adds a skeleton datastructure for electric vehicle (EV). Right now it is not much more than an EV charging point with a destinatio. In time it will include user preferences such as wait-time and so on. 

Closes #4  and #8.

Requires #10.

### CODE REVIEW

**Comment by mdavezac** [comment]
@cc-a, this is ready for review.

**Comment by mdavezac** [comment]
@cc-a , this is ready for review. Do note the todos github action :)

File: src/evosim/electric_vehicles.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+from enum import Enum, auto
+from pathlib import Path
+from typing import Optional, Sequence, Tuple, Union
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+
+from evosim.supply import LONDON_LATITUDE, LONDON_LONGITUDE, Chargers, Sockets
```
Comment (cc-a) [inline src/evosim/electric_vehicles.py:9]: supply.py no longer seems like the logical place for the lat and long data.

File: src/evosim/electric_vehicles.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        charger_distribution,
+        seed=rng,
+    )
+
+    result["dest_lat"] = rng.uniform(
+        high=np.max(latitude), low=np.min(latitude), size=n
+    )
+    result["dest_long"] = rng.uniform(
+        high=np.max(longitude), low=np.min(longitude), size=n
+    )
```
Comment (cc-a) [inline src/evosim/electric_vehicles.py:116]: Just for my clarity, the destination data here refers to the notional end point of the vehicles journey and not to a charging point assigned to the vehicle?

File: src/evosim/electric_vehicles.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    4     51.39      -0.45         CHADEMO    SLOW     51.37   1.18e+00
+    <BLANKLINE>
+                        model
+    0               BMW_225XE
+    1           TESLA_MODEL_X
+    2           KIA_NIRO_PHEV
+    3             NISSAN_LEAF
+    4  VOLVO_XC90_TWIN_ENGINE
+
+Much as the chargers and sockets, the models a categorical array taking their values
```
Comment (cc-a) [inline src/evosim/electric_vehicles.rst]:
```suggestion
Much as the chargers and sockets, the models are a categorical array taking their values
```

**Review by cc-a — APPROVED** [review]
Looks good to me.

File: src/evosim/electric_vehicles.py (hunk 2 above)
Comment (mdavezac) [reply src/evosim/electric_vehicles.py:116]: Yes

File: src/evosim/electric_vehicles.py (hunk 1 above)
Comment (mdavezac) [reply src/evosim/electric_vehicles.py:9]: Agreed. Now in #19.

**Pull request merged**